riscv/processor.cc

```diff
@@ -392,6 +392,9 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   csrmap[CSR_TDATA2] = tdata2 = std::make_shared<tdata2_csr_t>(proc, CSR_TDATA2);
   csrmap[CSR_TDATA3] = std::make_shared<tdata3_csr_t>(proc, CSR_TDATA3);
   csrmap[CSR_TINFO] = std::make_shared<tinfo_csr_t>(proc, CSR_TINFO);
+  csrmap[CSR_SCONTEXT] = std::make_shared<const_csr_t>(proc, CSR_SCONTEXT, 0);
+  csrmap[CSR_HCONTEXT] = std::make_shared<const_csr_t>(proc, CSR_HCONTEXT, 0);
+  csrmap[CSR_MCONTEXT] = std::make_shared<proxy_csr_t>(proc, CSR_MCONTEXT, csrmap[CSR_HCONTEXT]);
```

Comment:
The `csrmap` is intended for `csrrw` instructions that have only a CSR number and not a name. For internal code that accesses CSRs, prefer keeping a handle to it and referring to it by that handle.

The handle can be either local to this `reset()` method, such as `minstreth`, or if wider access is needed, in `state_t`, such as `mideleg`. It can be generic type `csr_t_p`, or if access to special methods are needed, a child class pointer.